File: Data Structures & Algorithms/duplicate-integer/submission-3.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        seen = {}

        for i in nums:
            if i in seen.keys():
                seen[i] += 1
            else:
                seen[i] = 1
        print(seen)

        for i in seen.values():
            if i > 1:
                return True

        return False

File: Data Structures & Algorithms/two-integer-sum/submission-3.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        Map<Integer,Integer> seen = new HashMap<Integer,Integer>();
        int[] response = new int[2];



        for (int i = 0; i < nums.length; i++) {
            if (seen.containsKey(target - nums[i])) {
                
                response[0] = seen.get(target-nums[i]);
                response[1] = (i);
                return response;
            } else {
                seen.put(nums[i], i);
            }
        }
        return response;

    }
    
}
